jetty-webapp/src/main/java/org/eclipse/jetty/webapp/Configurations.java

```diff
@@ -399,7 +399,7 @@ private void addConfiguration(Configuration configuration)
             {
                 Configuration c=i.next();
                 if(c.getClass().getName().equals(replaces.getName()) 
-                || c.replaces()!=null && c.replaces().getName().equals(replaces.getName()))
+                        || c.replaces()!=null && c.replaces().getName().equals(replaces.getName()))
                 {
                     i.set(configuration);
                     return;
```

Comment:
You can no longer just do a set and return here, because the new configuration may be replaced by one that is already present.    I think you need to just do i.remove()

```suggestion
                    i.remove();
```

Reply:
Luckily I don't like bananas.